src/redux/blockchain/blockchainActions.js

```diff
@@ -49,16 +50,17 @@ export const connect = () => {
     });
     const CONFIG = await configResponse.json();
     const { ethereum } = window;
-    const metamaskIsInstalled = ethereum && ethereum.isMetaMask;
-    if (metamaskIsInstalled) {
+    const provider = await detectEthereumProvider();
```

Comment:
`Web3.givenProvider` should be used instead of the metamask utility.

Reply:
This works too, thanks for comment.

Reply:
Not sure what you mean by "this works too". This could refer to the original code you posted, or this could refer to my comment.

The `detectEthereumProvider` would actually break the app because you are removing the check to make sure that the provider is metamask. If you read the documentation, the method returns any Ethereum provider it finds. If you only want metamask you need to pass in a specific option ([see docs](https://github.com/MetaMask/detect-provider#optionsmustbemetamask)).

In your personal code, sure do whatever you want if it works for you and gets the job done. I would just recommend that if you are trying to merge a pull request into open source you should try to avoid adding unnecessary packages, especially when the current packages already have methods that accomplish what is needed. In my opinion `"@metamask/detect-provider"` just appears to be unnecessary and using `Web3.givenProvider` or `window.ethereum` should be preferred.

Reply:
Sorry for the ambiguity, I meant to say it would be better to use `Web3.givenProvider`.